unified-runtime/source/adapters/offload/enqueue.cpp

```diff
@@ -107,17 +109,30 @@ ur_result_t doMemcpy(ur_command_t Command, ur_queue_handle_t hQueue,
 
   ol_event_handle_t EventOut = nullptr;
 
-  OL_RETURN_ON_ERR(olMemcpy(hQueue->OffloadQueue, DestPtr, DestDevice, SrcPtr,
-                            SrcDevice, size, phEvent ? &EventOut : nullptr));
-
+  ol_queue_handle_t Queue;
   if (blocking) {
-    OL_RETURN_ON_ERR(olSyncQueue(hQueue->OffloadQueue));
+    // If we are using a blocking operation, create a temporary queue that lives
+    // only for this function
+    OL_RETURN_ON_ERR(olCreateQueue(hQueue->OffloadDevice, &Queue));
```

Comment:
Technically this is only needed if `blocking` is true *and* `phEvent` is non-null, since we can just pass a null queue to `olMemcpy` and it will internally use one of the pooled queues

Reply:
Given the interface is changing in https://github.com/llvm/llvm-project/pull/150217 , I think it makes sense to do this change as part of the porting work for that or just after it.

Reply:
Actually, wait. That doesn't really affect this. One sec.